### PR TITLE
Document contributing guidelines

### PR DESCRIPTION
**Pull request by mristin**
* Wrote contributing guidelines in the docs

* Added coverage to test dependencies so that the developers can
  replicate the remote CI locally.

### CODE REVIEW

**Comment by mristin** [comment]
Fixes #37.

**Comment by mristin** [comment]
@maxfischer2781 The pull request is ready for review.

**Comment by mristin** [comment]
@maxfischer2781 just a gentle ping in case you forgot to review this pull request :-).

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+Here is an example of a commit message:
+
+.. code-block::
+
+    Add itertools.pairwise
+
+    * added itertools.pairwise
+
+    * documented itertools.pairwise
```
Comment (maxfischer2781) [inline docs/source/contributing.rst:159]: Can you please change that? This is basically the opposite of what I prefer: many small commits, each with one change. There shouldn't actually be a need to have a body for a commit message – that's a sure sign the commit is too large.  
The singular commit + summary is created when merging the branch into master.

Quick sketch of a possible description:
```suggestion
Make a single commit for each self-contained change.

Write the subject of the commit message in `imperative mood`_.
The subject should not exceed 50 characters.

Here is an example of two commit messages:

.. code-block::

    added itertools.pairwise

    documented itertools.pairwise
```

**Review by maxfischer2781 — CHANGES_REQUESTED** [review]
Thanks *a lot* for taking the time for this extensive writeup!

I do have one point that should be changed, though.

File: docs/source/contributing.rst (hunk 1 above)
Comment (mristin) [reply docs/source/contributing.rst:159]: @maxfischer2781 the current commit history uses present tense for the subject, while you used past tense in your examples here. What should be official?

Comment (maxfischer2781) [reply docs/source/contributing.rst:159]: Going through the recent commit history, the largest bulk uses *past tense* for the commit messages and *present tense* for PR messages. So, commits should be *past tense*.

Comment (mristin) [reply docs/source/contributing.rst]: @maxfischer2781 thanks! I changed the text and pushed another commit.

**Comment by mristin** [comment]
@maxfischer2781 please have another look, the new commit is review-ready.

**Comment by maxfischer2781** [comment]
LGTM. Again, thanks a lot for your contributions. 👍

**Pull request merged**